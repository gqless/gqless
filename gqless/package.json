{
  "license": "MIT",
  "name": "gqless",
  "version": "0.0.1-alpha.25",
  "sideEffects": false,
  "main": "dist/index.js",
  "source": "src/index.ts",
  "module": "dist/gqless.esm.js",
  "homepage": "https://gqless.netlify.com",
  "repository": {
    "type": "git",
    "url": "https://github.com/samdenty/gqless.git",
    "directory": "gqless"
  },
  "typings": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "watch": "tsdx watch",
    "build": "cp ../README.md .; tsdx build",
    "test": "jest"
  },
  "dependencies": {
    "@gqless/utils": "^0.0.1-alpha.25",
    "json-stable-stringify": "^1.0.1",
    "mix-classes": "^1.1.10",
    "tslib": "^1.10.0"
  },
  "devDependencies": {
    "@babel/core": "^7.6.4",
    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.4.4",
    "@types/json-stable-stringify": "^1.0.32",
    "template-literal-table": "^2.0.4"
  },
  "gitHead": "7c27436fcea8e672f07233010401417ea7c59760"
}
