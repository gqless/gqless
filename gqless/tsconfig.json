{
  "extends": "../tsconfig.json",
  "include": ["src"],
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "dist",
    // https://github.com/parcel-bundler/parcel/issues/2856
    "downlevelIteration": true
  },
  "references": [{ "path": "../packages/utils" }]
}
