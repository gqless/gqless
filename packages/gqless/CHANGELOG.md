# gqless

## 2.0.5

### Patch Changes

- 6fef085: add cache persistence helpers

## 2.0.4

### Patch Changes

- 2bf4ce2: add inner scheduler "getResolvingPromise"

## 2.0.3

### Patch Changes

- 27f9ece: set "graphql" as optional peerDependency

## 2.0.2

### Patch Changes

- c06ef80: add "prepass" helper

## 2.0.1

### Patch Changes

- a57cab4: official beta v2 publish
