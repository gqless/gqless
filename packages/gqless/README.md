# [GQless](https://gqless.com/getting-started) [![Documentation](https://img.shields.io/badge/documentation-documentation?color=C00B84)](https://gqless.com) [![Twitter](https://img.shields.io/twitter/follow/gqlessdev?label=%40gqlessdev&style=flat&logo=twitter&color=00acee)](https://twitter.com/gqlessdev)

> GQless is a fully-featured GraphQL client, which lowers the barrier of entry towards using GraphQL.

Data requirements within your application are picked up automatically, freeing you from having to maintain GraphQL queries by-hand. It offers a first-class TypeScript experience. See API documentation at all times within autocomplete.

Make breaking changes to your API, and see type-errors exactly where things are breaking, in realtime. No more running a separate validation step.

**More documentation is available at [gqless.com/getting-started](https://gqless.com/getting-started).**

<a href="https://gqless.com">
  <img alt="Example usage" src="https://user-images.githubusercontent.com/13242392/112103674-fddc4980-8ba1-11eb-8c83-b527dcb0243d.PNG" />
</a>
