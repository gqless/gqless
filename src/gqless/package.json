{
  "name": "gqless",
  "version": "0.0.1-alpha.19",
  "main": "dist/index.js",
  "module": "dist/gqless.esm.js",
  "homepage": "https://github.com/samdenty/gqless",
  "typings": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "watch": "tsdx watch",
    "build": "cp ../../README.md .; tsdx build --tsconfig ./tsconfig.build.json",
    "test": "tsdx test"
  },
  "jest": {
    "snapshotSerializers": [
      "<rootDir>/../../jest/snapshotSerializer.ts"
    ],
    "moduleNameMapper": {
      "@gqless/(.*)$": "<rootDir>/../../packages/$1/src",
      "gqless": "<rootDir>/src"
    }
  },
  "dependencies": {
    "@gqless/utils": "^0.0.1-alpha.19",
    "json-stable-stringify": "^1.0.1",
    "lodash": "^4.17.13",
    "mix-classes": "^1.1.7"
  },
  "devDependencies": {
    "@babel/core": "^7.4.4",
    "@types/json-stable-stringify": "^1.0.32",
    "@types/lodash": "^4.14.136",
    "template-literal-table": "^2.0.2"
  },
  "gitHead": "7c27436fcea8e672f07233010401417ea7c59760"
}
