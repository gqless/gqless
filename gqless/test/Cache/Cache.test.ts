import { Cache, Value, merge, StaticExtension, GET_KEY } from 'gqless'
import { schema } from '@internal/fixtures'

let cache: Cache
beforeEach(() => {
  cache = new Cache(schema.Query)
})

it('records instances', () => {
  const value = new Value(schema.String)
  const value2 = new Value(schema.String)

  expect(cache.entries.size).toBe(1)

  cache.rootValue.set('a', value)
  expect(cache.entries.size).toBe(2)
  const entry = cache.entries.get(value.node)!
  expect(entry.instances.has(value)).toBeTruthy()

  cache.rootValue.set('a', value2)
  expect(entry.instances.has(value)).toBeFalsy()
  expect(entry.instances.has(value2)).toBeTruthy()
})

it('supports matching', () => {
  const aValue = new Value(schema.String, 'test-a')
  const bValue = new Value(schema.String, 'test-b')

  cache.rootValue.set('a', aValue)
  cache.rootValue.set('b', bValue)

  const stringEntry = cache.entries.get(schema.String)!

  expect(stringEntry.match(/test-/)!.value).toBe(aValue)
  expect(stringEntry.match(/b/)!.value).toBe(bValue)
})

it('supports merging', () => {
  merge(cache, cache.rootValue, { object: { int: 1 } })

  expect(cache.toJSON().data).toMatchInlineSnapshot(`
    Object {
      "__typename": "Query",
      "object": Object {
        "__typename": "Object",
        "int": 1,
      },
    }
  `)

  merge(cache, cache.rootValue, { object: { string: 'test' } })

  expect(cache.toJSON().data).toMatchInlineSnapshot(`
    Object {
      "__typename": "Query",
      "object": Object {
        "__typename": "Object",
        "int": 1,
        "string": "test",
      },
    }
  `)

  merge(cache, cache.rootValue, { object: null })

  expect(cache.toJSON().data).toMatchInlineSnapshot(`
    Object {
      "__typename": "Query",
      "object": null,
    }
  `)
})

it('supports keys', () => {
  const queryExtensions = [
    new StaticExtension(undefined, schema.Query, {
      object: {
        [GET_KEY]: (obj: any) => obj.int,
      },
      object2: {
        [GET_KEY]: (obj: any) => obj.int,
      },
    }),
  ]

  merge(
    cache,
    cache.rootValue,
    { object: { int: 1, string: 'a' } },
    queryExtensions
  )

  const objectKey = cache.entries.get(schema.Object)!
  const keyedValue = objectKey.getByKey(1)!

  expect(keyedValue.toJSON()).toMatchInlineSnapshot(`
    Object {
      "__typename": "Object",
      "int": 1,
      "string": "a",
    }
  `)

  merge(
    cache,
    cache.rootValue,
    { object2: { int: 1, string2: 'b' } },
    queryExtensions
  )

  expect(cache.rootValue.get('object')).toBe(cache.rootValue.get('object2'))

  expect(keyedValue.toJSON()).toMatchInlineSnapshot(`
    Object {
      "__typename": "Object",
      "int": 1,
      "string": "a",
      "string2": "b",
    }
  `)

  expect(cache.toJSON()).toMatchSnapshot()
})
