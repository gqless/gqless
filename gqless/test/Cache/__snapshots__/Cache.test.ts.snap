// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supports keys 3`] = `
Object {
  "data": Object {
    "__typename": "Query",
    "object": Object {
      "__typename": "Object",
      "int": 1,
      "string": "a",
      "string2": "b",
    },
    "object2": Object {
      "__typename": "Object",
      "int": 1,
      "string": "a",
      "string2": "b",
    },
  },
  "types": Object {
    "Int": Object {
      "instances": Array [
        1,
      ],
      "keys": Object {},
    },
    "Object": Object {
      "instances": Array [
        Object {
          "__typename": "Object",
          "int": 1,
          "string": "a",
          "string2": "b",
        },
      ],
      "keys": Object {
        "1": Object {
          "__typename": "Object",
          "int": 1,
          "string": "a",
          "string2": "b",
        },
      },
    },
    "Query": Object {
      "instances": Array [
        Object {
          "__typename": "Query",
          "object": Object {
            "__typename": "Object",
            "int": 1,
            "string": "a",
            "string2": "b",
          },
          "object2": Object {
            "__typename": "Object",
            "int": 1,
            "string": "a",
            "string2": "b",
          },
        },
      ],
      "keys": Object {},
    },
    "String": Object {
      "instances": Array [
        "a",
        "b",
      ],
      "keys": Object {},
    },
  },
}
`;
